src/python_testing/TC_DeviceConformance.py

```diff
@@ -125,6 +141,13 @@ def record_warning(location, problem):
                 for f in feature_masks:
                     location = AttributePathLocation(endpoint_id=endpoint_id, cluster_id=cluster_id,
                                                      attribute_id=GlobalAttributeIds.FEATURE_MAP_ID)
+                    if cluster_id == Clusters.AccessControl.id and f == Clusters.AccessControl.Bitmaps.Feature.kManagedDevice:
+                        # Managed ACL is treated as a special case because it is only allowed if other endpoints support NIM and disallowed otherwise.
+                        if not self._has_nim():
```

Comment:
Given that we will likely add to this list in the future, shouldn't this be self._has_allowed_MACL()?

Do we have hard coding of the device library rules in the other scripts?